test/connection/transaction_test.exs

```diff
@@ -0,0 +1,200 @@
+    std_offset: 0,
+    time_zone: "Etc/UTC"
+  }
+
+  setup do
+    table = Transaction.init_transaction_call_table()
+    transaction = Transaction.generate_transaction!(table)
+
+    [table: table, transaction: transaction]
+  end
+
+  describe "Handles response for up to date request" do
+    setup %{table: table, transaction: transaction} do
+      Transaction.insert_transaction(table, transaction, from(), 10000)
+      []
+    end
+
+    test "when result is a success", %{table: table, transaction: transaction} do
+      response = {:ok, %{"data" => "example"}}
+
+      logs =
+        capture_log(fn ->
+          Transaction.handle_transaction(response, transaction, table)
+        end)
+
+      assert_receive {@tag, ^response}
+      assert logs =~ "Call OK:"
+    end
+
+    test "when response is an error", %{table: table, transaction: transaction} do
+      response = {:error, 418, "reason"}
+
+      logs =
+        capture_log(fn ->
+          Transaction.handle_transaction(response, transaction, table)
+        end)
+
+      assert_receive {@tag, ^response}
+      assert logs =~ "Call ERROR:"
+    end
+  end
+
+  describe "Handles response when" do
+    test "request is outdated", %{table: table, transaction: transaction} do
+      # Deliberetely inserting outdated record
+      :ets.insert(table, {transaction, from(), 0})
+
+      logs =
+        capture_log(fn ->
+          Transaction.handle_transaction({:ok, %{}}, transaction, table)
+        end)
+
+      refute_receive _
+      assert logs =~ "Received OK reply to the outdated call"
+    end
+
+    test "when request is unkown", %{table: table, transaction: transaction} do
+      logs =
+        capture_log(fn ->
+          Transaction.handle_transaction({:ok, %{}}, transaction, table)
+        end)
+
+      refute_receive _
+      assert logs =~ "Received OK reply to the unknown call"
+    end
+  end
+
+  describe "insert_transaction adds new transaction" do
+    test "with proper expiration time", %{table: table, transaction: transaction} do
+      with_mock DateTime, [:passthrough], utc_now: fn -> @now end do
+        Transaction.insert_transaction(table, transaction, from(), @timeout)
+
+        expires =
+          @now
+          |> DateTime.add(@timeout, :millisecond)
+          |> DateTime.to_unix(:millisecond)
+
+        case :ets.lookup(table, transaction) do
+          [{_, _, expires_at}] ->
+            assert expires == expires_at
+
+          _ ->
+            raise "Transaction has not been added!"
+        end
+      end
+    end
+  end
+
+  describe "transaction_status returns" do
+    setup_with_mocks(
+      [{DateTime, [:passthrough], utc_now: fn -> @now end}],
+      %{table: table, transaction: transaction}
+    ) do
+      Transaction.insert_transaction(table, transaction, from(), @timeout)
+      []
+    end
+
+    test ":ok tuple if transaction is up to date", %{table: table, transaction: transaction} do
+      with_mock DateTime, [:passthrough],
```

Comment:
Mocking modules globally is discouraged  in Elixir. If you really need to, then at least you need to add `async: flase` when using `ExUnit.Case`, but I'd consider using  [`mox`](https://github.com/dashbitco/mox) or [`mockery`](https://github.com/appunite/mockery) instead

Reply:
[Docs](https://hexdocs.pm/ex_unit/ExUnit.Case.html) says that default value is `async: false`. I have put too much effort to find equivalent functionality in metioned `mox` or `mockery` - I am proposing to stick with sync tests